### PR TITLE
RavenDB-17542 - Add ForcePathStyle to S3 Settings

### PR DESCRIPTION
**Pull request by grisha-kotler**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-17542

### Additional description

Add `ForcePathStyle` to S3 Settings (setting to true for backward compatibility)

### Type of change

- Bug fix

### How risky is the change?

- Low 

### Backward compatibility

- Ensured. The same as it was prior to using the official S3 client.

### Documentation update

- This change requires a documentation update.

### Testing 

- It has been verified by manual testing

### UI work

- Added the field in the UI

### CODE REVIEW

File: src/Raven.Studio/wwwroot/App/views/partial/backupDestinations.html [hunk 1]
```diff
@@ -118,7 +118,14 @@ <h3>Destination</h3>
                         <div data-bind="template: { name: 'backup-via-configuration-script' }"></div>
                     </div>
                     <div data-bind="if: !hasConfigurationScript()">
-                        
+                        <div class="row margin-bottom" data-bind="collapse: useCustomS3Host">
+                            <div class="col-xs-offset-12 col-sm-offset-6 col-lg-offset-4">
+                                <div class="toggle margin-left">
+                                    <input id="forcePathStyle" type="checkbox" data-bind="checked: forcePathStyle">
```
Comment (ml054) [inline src/Raven.Studio/wwwroot/App/views/partial/backupDestinations.html:124]: can we include some tooltip what's that, do I need this etc?

Comment (ml054) [reply src/Raven.Studio/wwwroot/App/views/partial/backupDestinations.html:124]: image

File: src/Raven.Client/Documents/Operations/Backups/BackupSettings.cs [hunk 2]
```diff
@@ -145,6 +145,8 @@ public class S3Settings : AmazonSettings
         /// </summary>
         public string CustomServerUrl { get; set; }
 
+        public bool ForcePathStyle { get; set; } = true;
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Operations/Backups/BackupSettings.cs]: This is by default `false` (https://github.com/aws/aws-sdk-net/blob/60ad7e792a5c32423297f13df9e0b40a1ca510e5/sdk/src/Services/S3/Custom/AmazonS3Config.cs#L37)

Please explain why we want to set it to true by default.

Comment (grisha-kotler) [reply src/Raven.Client/Documents/Operations/Backups/BackupSettings.cs]: To keep backward compatibility with the previous version, when we didn't use the official client

Comment (ppekrol) [reply src/Raven.Client/Documents/Operations/Backups/BackupSettings.cs]: but what is the implication of setting that to true for everyone else?

File: src/Raven.Studio/wwwroot/App/views/partial/backupDestinations.html (hunk 1 above)
Comment (grisha-kotler) [reply src/Raven.Studio/wwwroot/App/views/partial/backupDestinations.html:124]: done

File: src/Raven.Client/Documents/Operations/Backups/BackupSettings.cs (hunk 2 above)
Comment (grisha-kotler) [reply src/Raven.Client/Documents/Operations/Backups/BackupSettings.cs]: done

File: src/Raven.Studio/wwwroot/App/views/resources/createDatabase.html [hunk 3]
```diff
@@ -477,6 +477,14 @@ <h3>Backup Source Configuration</h3>
                        </div>
                    </div>
                </div>
+               <div class="row margin-bottom" data-bind="collapse: useCustomS3Host">
```
Comment (ml054) [inline src/Raven.Studio/wwwroot/App/views/resources/createDatabase.html:480]: can you verify that it should be collapsed when using custom s3 host?

Comment (grisha-kotler) [reply src/Raven.Studio/wwwroot/App/views/resources/createDatabase.html:480]: We only need this when using a custom URL.
When using Amazon AWS, the default value will always work.

Comment (ml054) [reply src/Raven.Studio/wwwroot/App/views/resources/createDatabase.html:480]: ok thanks for confirming that.

**Review by ml054 — APPROVED** [review]

**Pull request merged**